editor_canvas.py

```
@@ -2,46 +2,62 @@
 from tkinter import messagebox
 from enum import Enum, auto
 import re
-import rendering
-from typing import Any
+import math
+from typing import Any, Tuple, List, Optional
```

Ruff (0.15.6)

[warning] `typing.Tuple` is deprecated, use `tuple` instead

(UP035)